state/core_access.go

```diff
@@ -153,7 +153,7 @@ func (ca *CoreAccessor) SubmitPayForData(
 ) (*TxResponse, error) {
 	response, err := payment.SubmitPayForData(ctx, ca.signer, ca.coreConn, nID, data, gasLim)
 	// metrics should only be counted on a successful PFD tx
-	if response.Code == 0 && err == nil {
+	if err == nil && response.Code == 0 {
```

Comment:
```suggestion
	if err == nil && response != nil && response.Code == 0 {
```

Reply:
This is more semantic as it reads:

1. if there is no error
2. and we have an actual response
3. and the response Code is 0

Reply:
If I would design the API it would be just an error from which you can get an error code if needed. 3 conditions after doing a call is 😬